Backend/Gmfctn/Controllers/UserController.cs

```diff
@@ -1,39 +1,36 @@
-﻿using AutoMapper;
-using Data_;
-using Data_.Dtos;
+﻿using Data_.Dtos;
 using Data_.Entities;
-using Data_.Interfaces;
-using Data_.Validators;
+using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.EntityFrameworkCore;
+using Services.Interfaces;
 using System;
 using System.Collections.Generic;
-using System.Linq;
-using System.Text.RegularExpressions;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace Gmfctn.Controllers
 {
     [Route("api/user")]
     [ApiController]
+    [Authorize]
+
     public class UserController : ControllerBase
     {
-        private IUnitOfWork UnitOfWork;
-        private readonly IMapper _Mapper;
-        
-        public UserController(GmfctnContext Context, IMapper Mapper)
+        private readonly IUserService UserService;
+        public UserController(IUserService _UserService)
         {
-            UnitOfWork = new UnitOfWork(Context);
-            _Mapper = Mapper;
+            UserService = _UserService;
         }
-        [HttpGet]
-        public async Task<ActionResult<IEnumerable<User>>> GetAllUsers(CancellationToken Cancel)
+
+        [HttpGet("get_all_users")]
+        [Authorize(Roles = "Admin")]
+        public async Task<ActionResult<IEnumerable<UserReadDTO>>> GetAllUsers(CancellationToken Cancel)
         {
-           
+
             try
             {
-                var Users = await UnitOfWork.UserRepository.GetAll(Cancel);
+                var Users = await UserService.GetAllUsers(Cancel);
```

Comment:
The same as above about handling errors and return from Service logic

Reply:
Fixed